### PR TITLE
row: allow title to be truncated to (n) lines

### LINKED ISSUE #497: row: add prop to truncate title lines
### Desired behavior

As a developer, I can add a prop to truncate the title to a defined a number of lines. Title has ellipses for overflow text.

image

### Today's behavior

Title wraps indefinitely

### The Value Add

Allows for uniformity in dense row layouts.

### The Tradeoffs

Yet another prop.

### PR DESCRIPTION
**Pull request by danethurber**
i had to do some awkward things to make this a non-breaking change. let's discuss

resolves #497 

Outstanding
- [x] docs

### CODE REVIEW

File: packages/row/package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "@pluralsight/ps-design-system-prop-types": "^2.0.0",
     "@pluralsight/ps-design-system-theme": "^2.0.0",
     "@pluralsight/ps-design-system-util": "^2.0.0",
-    "prop-types": "^15.5.10"
+    "prop-types": "^15.7.2",
+    "shave": "^2.5.3"
```
Comment (jaketrent) [inline packages/row/package.json]: nice and lightweight.  seems like a good choice.

File: packages/row/src/react/shave.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    this.el.innerHTML = this.props.children
+  }
+
+  truncate() {
+    const maxHeight = this.props.lineHeight * this.props.lines
+    shave(this.el, maxHeight, { character: this.props.character })
+  }
+
+  render() {
+    return <div ref={this.bindRef} {...filterReactProps(this.props)} />
```
Comment (jaketrent) [inline packages/row/src/react/shave.js]: Do we want to use `React.createRef()` instead?

File: packages/row/src/react/shave.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import shave from 'shave'
+
+import core from '@pluralsight/ps-design-system-core'
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+
+class Shave extends React.Component {
```
Comment (jaketrent) [inline packages/row/src/react/shave.js]: I love your react wrapper implementation around Shave.

**Comment by danethurber** [comment]
change to be true/false is truncated. truncate to 2 lines

**Pull request merged**